authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,734 @@
+
+. The `sts-preflight` tool does not put conditions on the IAM roles it creates to allow only specific `ServiceAccount` objects to assume a role. To restrict which identities can assume which role, you must manually modify the trust relationship of each role to add a corresponding condition.
++
+[NOTE]
+====
+Role names that exceed 64 characters might be truncated.
+====
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-image-registry-installer-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-image-registry:registry",
+      "system:serviceaccount:openshift-image-registry:cluster-image-registry-operator"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-ingress-operator-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-ingress-operator:ingress-operator"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-cluster-csi-drivers-ebs-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-cluster-csi-drivers:aws-ebs-csi-driver-operator",
+      "system:serviceaccount:openshift-cluster-csi-drivers:aws-ebs-csi-driver-controller-sa"
+    ]
+  }
+}
+----
+
+.. Modify the trust relationship of the `<aws-infra-name>-openshift-machine-api-aws-cloud-credentials` role to have the following condition:
++
+[source,json]
+----
+"Condition": {
+  "StringEquals": {
+    "s3.us-east-1.amazonaws.com/<aws-infra-name>-installer:sub": [
+      "system:serviceaccount:openshift-machine-api:machine-api-controllers"
+    ]
+  }
+}
+----
+
+. Copy the manifests that the `sts-preflight` tool created to the `manifests` directory in the same location as the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ cp _output/manifests/* <path-to-install-config.yaml-dir>/manifests/
+----
+
+. Copy the private key for the `ServiceAccount` object signer to the same location as the `install-config.yaml` file:
++
+[source,terminal]
+----
+$ cp -a _output/tls <path-to-install-config.yaml-dir>
+----
+
+. Create a token that is trusted by the OIDC identity provider:
++
+[source,terminal]
+----
+$ ./sts-preflight token
+----
++
+This token is used to assume a role to get the credentials that are required to start the installation. The token is created in the `_output/token` directory.
+
+. Use the token to assume the installer role and mint the credentials that are required for the {product-title} installation process.
```

Comment:
Will folks know what _mint the credentials_ means? All I see on Google is links to Mint Mode docs and Linux Mint.

Reply:
s/mint/create :)